src/devices/src/virtio/block/device.rs

```diff
@@ -484,9 +489,69 @@ pub(crate) mod tests {
             &mut EventManager::new().unwrap(),
         );
         // Validate the queue operation finished successfully.
+        #[cfg(not(fuzzing))]
         assert_eq!(b.interrupt_evt.read().unwrap(), 1);
     }
 
+    #[cfg(fuzzing)]
+    pub fn fuzzing_func(data: &[u8]) {
+        const DESC_DATA_SIZE: usize = 5;
+        const VQUEUE_SIZE: usize = 16;
+        let mut block = default_block();
+        let mem = default_mem();
+        let vq = VirtQueue::new(GuestAddress(0), &mem, VQUEUE_SIZE as u16);
+        let mut idx = 0;
+        let mut data_size = data.len();
+        let mut rdr = Cursor::new(data);
+
+        block.set_queue(0, vq.create_queue());
+        block.activate(mem.clone()).unwrap();
+        initialize_virtqueue(&vq);
+
+        loop {
+            if data_size < DESC_DATA_SIZE {
+                break;
+            }
+
+            let addr = rdr.read_u16::<LittleEndian>().unwrap_or_default();
+            let mut len = rdr.read_u8().unwrap_or_default();
+            let flags = rdr.read_u8().unwrap_or_default();
+            let next = rdr.read_u8().unwrap_or_default();
+
+            data_size -= DESC_DATA_SIZE;
+
+            if len as usize > data_size {
+                len = data_size as u8;
+            }
+
+            vq.avail.ring[idx].set(idx as u16);
+            vq.dtable[idx].set(addr as u64,
+                               len as u32,
+                               flags as u16,
+                               next as u16);
+
+            let mut bytes_written: usize = 0;
+            for i in addr..=0xffff {
+                if i - addr >= (len as u16) {
+                    break;
+                }
+
+                let data_byte = rdr.read_u8().unwrap_or_default();
+                mem.write_obj::<u8>(data_byte, GuestAddress(i as u64));
+                bytes_written += 1;
+            }
+
+            data_size -= bytes_written;
+            idx += 1;
+            if idx >= VQUEUE_SIZE {
+                break;
+            }
+        }
+
+        vq.avail.idx.set(1);
+        invoke_handler_for_queue_event(&mut block);
```

Comment:
Here you can directly call 'block.process_queue(0)' instead to avoid operations you don't care about